### DISCUSSION TITLE
Issue after upgrading log4j 1.2.16 to log4j 1.2.17

### DISCUSSION THREAD

**Discussion by sal4455** [nextgenhealthcare/connect#5231] [Q&A]
Hi,
We are using MirthConnect version 3.11.0 on linux machine. We have recently upgraded log4j version to **log4j 1.2.17**  because of some vulnerabilities .
After upgrading ,Mirthconnect server is up and running but we are unable to login into MirthConnect Administrator .
I am attaching the screenshot of the error .
Kindly have a look and help us resolving this.

![image (4)](https://user-images.githubusercontent.com/104156028/172600114-0095c9df-fa1c-4cc6-b8aa-3c6b68b7de9a.png)

**Comment by jonbartels** [comment]
https://github.com/kpalang/mirth-plugin-guide/#pay-attention-here explains what's going on.

The JARs used by the MC client need to be signed. You can truly solve the issue by getting a code signing certificate and signing `log4j-1.2.17.jar` with your cert.

The workaround is to sign the `log4j-1.2.17.jar` with a self-signed cert then run the MC launcher with the `-k` flag.

**Reply by sal4455** [reply]
Thankyou for the response,
  We tried above approach .
We also tried reinstalling MirthConnect 3.11.0 and then upgraded the log4j version to 1.2.17 inside each folders again.
Now when starting mcserver by "./mcserver start"  or  "java -jar mirth-server-launcher.jar" .
This is showing diff error , I attached ss of the error .
Please look into it and guide us .

![15bcf51b-a66d-40ab-92f7-9156cf7a8f63](https://user-images.githubusercontent.com/104156028/173035632-061cc71b-8e9e-4276-b4ff-08cb34534d01.png)

**Reply by jonbartels** [reply]
That looks like a typical classpath issue. 

What is the path to `log4j-1.2.17.jar`? Is it in `server-lib`?

**Reply by sal4455** [reply]
yes the log4j-1.2.17.jar is in server-lib, cli-lib, client-lib and manager-lib

**Reply by ab-20-m** [reply]
The MirthLauncher class specifically looks for the log4j as [log4j-1.2.16.jar](https://github.com/nextgenhealthcare/connect/blob/74e70c466d1ec72e19ae020e893210a40f037d9b/server/src/com/mirth/connect/server/launcher/MirthLauncher.java#L45). You can use the [exe4j vmoptions file](https://www.ej-technologies.com/resources/exe4j/help/doc/help.pdf) and the -classpath/p option to prepend the log4j-1.2.17.jar to the classpath.

Or just rename the file to log4j-1.2.16.jar.
